newIDE/app/src/MainFrame/index.js

```diff
@@ -1408,9 +1408,20 @@ class MainFrame extends React.Component<Props, State> {
       .then(fileMetadata => {
         if (!fileMetadata) return;
 
-        return this.openFromFileMetadata(fileMetadata).then(() =>
-          this.openSceneOrProjectManager()
-        );
+        return this.openFromFileMetadata(fileMetadata).then(() => {
+          this.openSceneOrProjectManager();
+          const recentFiles = localStorage.getItem('gd-recent-files');
+          if (recentFiles) {
+            const parsedData = JSON.parse(recentFiles);
+            parsedData.push(fileMetadata.fileIdentifier);
```

Comment:
We should save the entire fileMetadata. This is because some storage providers like Google Drive or other in the futures might want to save something else than just a `fileIdentifier`.
So we need everything :) 

Right now, it's true that recent files will be only for the desktop app, but in the future we might show the menu in the web-app, in which case we want to remember all kind of file that was opened.